client/src/lib/ipfs/pinataIpfsClient.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import { IpfsClient } from "./ipfsClient";
+
+export class PinataIpfsClient extends IpfsClient {
+  private pinataJwt: string;
+
+  constructor(pinataJwt: string | undefined) {
+    super();
+    if (!pinataJwt) {
+      throw new Error("Pinata JWT is required");
+    }
+    this.pinataJwt = pinataJwt;
+  }
+
+  async read(hashOrCid: string): Promise<string | null> {
+    try {
+      if (hashOrCid.startsWith("0x")) {
+        hashOrCid = this.convertBytes32ToIpfsCid(hashOrCid);
+      }
+      const res = await axios.get(`https://gateway.pinata.cloud/ipfs/${hashOrCid}`);
```

Comment:
You could use `HEAD` request to get size. https://specs.ipfs.tech/http-gateways/path-gateway/#head-ipfs-cid-path-params